src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBarSetReminder.vue

```diff
@@ -0,0 +1,176 @@
+
+import NcActionButton from '@nextcloud/vue/dist/Components/NcActionButton.js'
+import NcPopover from '@nextcloud/vue/dist/Components/NcPopover.js'
+
+import { setMessageReminder } from '../../../../../services/remindersService.js'
+
+export default {
+	name: 'MessageButtonsBarSetReminder',
+
+	components: {
+		NcActionButton,
+		NcPopover,
+	},
+
+	props: {
+		/**
+		 * The conversation token.
+		 */
+		token: {
+			type: String,
+			required: true,
+		},
+
+		/**
+		 * The message id.
+		 */
+		id: {
+			type: [String, Number],
+			required: true,
+		},
+	},
+
+	computed: {
+		containerElement() {
+			return document.querySelector('.messages-list__scroller')
+		},
+
+		reminderOptions() {
+			const currentDateTime = moment()
+
+			// Same day 18:00 PM (or hidden)
+			const laterTodayTime = (currentDateTime.hour() < 18)
+				? moment().hour(18)
+				: null
+
+			// Tomorrow 08:00 AM
+			const tomorrowTime = moment().add(1, 'days').hour(8)
+
+			// Saturday 08:00 AM (or hidden)
+			const thisWeekendTime = (currentDateTime.day() !== 6 && currentDateTime.day() !== 0)
+				? moment().day(6).hour(8)
+				: null
+
+			// Next Monday 08:00 AM
+			const nextWeekTime = moment().add(1, 'weeks').day(1).hour(8)
+
+			return [
+				{
+					key: 'laterToday',
+					icon: Update,
+					timestamp: this.getTimeStamp(laterTodayTime),
+					label: t('spreed', 'Later today — {timeLocale}', { timeLocale: laterTodayTime?.format('ddd LT') }),
+					ariaLabel: t('spreed', 'Set reminder for later today'),
+				},
+				{
+					key: 'tomorrow',
+					icon: ChevronRight,
+					timestamp: this.getTimeStamp(tomorrowTime),
+					label: t('spreed', 'Tomorrow — {timeLocale}', { timeLocale: tomorrowTime?.format('ddd LT') }),
+					ariaLabel: t('spreed', 'Set reminder for tomorrow'),
+				},
+				{
+					key: 'thisWeekend',
+					icon: CalendarWeekend,
+					timestamp: this.getTimeStamp(thisWeekendTime),
+					label: t('spreed', 'This weekend — {timeLocale}', { timeLocale: thisWeekendTime?.format('ddd LT') }),
+					ariaLabel: t('spreed', 'Set reminder for this weekend'),
+				},
+				{
+					key: 'nextWeek',
+					icon: ChevronDoubleRight,
+					timestamp: this.getTimeStamp(nextWeekTime),
+					label: t('spreed', 'Next week — {timeLocale}', { timeLocale: nextWeekTime?.format('ddd LT') }),
+					ariaLabel: t('spreed', 'Set reminder for next week'),
+				},
+			].filter(option => option.timestamp !== null)
+		},
+	},
+
+	methods: {
+		getTimeStamp(momentObject) {
+			return momentObject?.minute(0).second(0).millisecond(0).valueOf() || null
+		},
+
+		async setReminder(timestamp) {
+			try {
+				await setMessageReminder(this.token, this.id, timestamp / 1000)
+				showSuccess(t('spreed', 'A reminder was successfully created'))
+			} catch (error) {
+				showError(t('spreed', 'Error occurred when creating a reminder'))
+			}
+		},
+	},
+}
+</script>
+
+<style lang="scss" scoped>
+.popover_nested-list {
+	:deep(.action-button::after) {
+		content: "\FF1E";
```

Comment:
The arrow right looks kind of weird to me. Not sure we need it. Also the files app has nothing:
https://github.com/nextcloud/server/issues/39181#issuecomment-1664832347

That being said, it looks like files is replacing the popup or something, instead of adding another one to the right.
But not into the details there

Reply:
Looks good enough like that, but opening to the right could be problematic?

Edit: ah seems to open to the left in the call case, but yeah the arrow would be wrong at that moment, so let's not have it for now:
![grafik](https://github.com/nextcloud/spreed/assets/213943/85cb8967-ec0a-4502-be00-6890bb71585e)

Reply:
What about icon from `var(--icon-triangle-e-light)`?

Reply:
> What about icon from `var(--icon-triangle-e-light)`?

![image](https://github.com/nextcloud/spreed/assets/93392545/4cd6e203-0e0a-4184-96e4-c813eb6af2d1)

Reply:
> Let's just not use any icon for now, because it could show into the wrong direction (as outlined above)

It seems to me to be a "default" behavior for multi-level context menus everywhere.

![image](https://github.com/nextcloud/spreed/assets/25978914/1afd260e-3e6b-4739-a1e0-c2964c6666af)

![image](https://github.com/nextcloud/spreed/assets/25978914/30fd9523-1167-4892-992a-7e39a374b328)